dev/tools/gen_keycodes/bin/gen_keycodes.dart

```diff
@@ -0,0 +1,160 @@
+        'If --android-keycodes is not specified, the input will be read from the '
+        'correct file in the Android repository.',
+  );
+  argParser.addOption(
+    'android-scancodes',
+    defaultsTo: null,
+    help: 'The path to where the Android scancodes header file should be read. '
+      'If --android-scancodes is not specified, the input will be read from the '
+      'correct file in the Android repository.',
+  );
+  argParser.addOption(
+    'android-domkey',
+    defaultsTo: path.join(flutterRoot.path, 'dev', 'tools', 'gen_keycodes', 'data', 'key_name_to_android_name.json'),
+    help: 'The path to where the Android keycode to DomKey mapping is.',
+  );
+  argParser.addOption(
+    'data',
+    defaultsTo: path.join(flutterRoot.path, 'dev', 'tools', 'gen_keycodes', 'data', 'key_data.json'),
+    help: 'The path to where the key code data file should be written when '
+        'collected, and read from when generating output code. If --data is '
+        'not specified, the output will be written to/read from the current '
+        "directory. If the output directory doesn't exist, it, and the path to "
+        'it, will be created.',
+  );
+  argParser.addOption(
+    'code',
+    defaultsTo: path.join(flutterRoot.path, 'packages', 'flutter', 'lib', 'src', 'services', 'keyboard_key.dart'),
+    help: 'The path to where the output "keyboard_keys.dart" file should be'
+        'written. If --code is not specified, the output will be written to the '
+        'correct directory in the flutter tree. If the output directory does not '
+        'exist, it, and the path to it, will be created.',
+  );
+  argParser.addOption(
+    'maps',
+    defaultsTo: path.join(flutterRoot.path, 'packages', 'flutter', 'lib', 'src', 'services', 'keyboard_maps.dart'),
+    help: 'The path to where the output "keyboard_maps.dart" file should be'
+      'written. If --maps is not specified, the output will be written to the '
+      'correct directory in the flutter tree. If the output directory does not '
+      'exist, it, and the path to it, will be created.',
+  );
+  argParser.addFlag(
+    'collect',
+    defaultsTo: false,
+    negatable: false,
+    help: 'If this flag is set, then collect and parse header files from '
+        'Chromium and Android instead of reading pre-parsed data from '
+        '"key_data.json", and then update "key_data.json" with the fresh data.',
+  );
+  argParser.addFlag(
+    'help',
+    defaultsTo: false,
+    negatable: false,
+    help: 'Print help for this command.',
+  );
+
+  final ArgResults parsedArguments = argParser.parse(rawArguments);
+
+  if (parsedArguments['help']) {
+    print(argParser.usage);
+    exit(0);
+  }
+
+  KeyData data;
+  if (parsedArguments['collect']) {
+    String hidCodes;
+    if (parsedArguments['chromium-hid-codes'] == null) {
+      hidCodes = await getChromiumConversions();
+    } else {
+      hidCodes = await File(parsedArguments['chromium-hid-codes']).readAsString();
```

Comment:
Might as well use the sync versions throughout.

Reply:
Sure.  Done.